src/components/ClauseComponent.js

```diff
@@ -81,8 +86,35 @@ function ClauseComponent(props) {
         .reduce((props, node) => ({ ...props, ...node }), {})
     });
 
+    let listData = {};
+    const findListVariables = (node) =>{
+      if(node.type === 'ul_list'){
+        (node.nodes || []).forEach((listNode,index) => {
+         if(listNode.getInlinesByType("variable").size > 0){
+            listData[listNode.key] = { 
+                currentText: listNode.text,
+                head: index === 0 ? true : false, 
+                tail: index === node.nodes.size - 1 ? true : false,
```

Comment:
I see `head` and `tail` are used to determine if this is a node which can be removed or an empty node with an add icon. Can you explain though the logic here on line 96 and 97, I'm not sure I understand it fully.

Reply:
Assuming there are several groups of variable lists in a clause, for each of these groups of lists, I needed to keep track of the first and last list item for each group, so while iterating through each list group, I keep track of the head by identifying the list item(a node) is at index 0, and keep track of the tail by identifying which list item is at the last node in each List group, by checking where index===" node.nodes.size - 1 " (i.e similar to saying array.length - 1). Therefore, in each list group, the head property of the first list item, will evaluate to true and the tail property of the last list item will evaluate to true. I now use these proeprties to ensure that the first list(head) can't be removed and the add icon, comes after the last node.

Reply:
AH okay yes I get it now. Small code style difference but I think this could be:

```js
head: index === 0,
tail: index === (node.nodes.size - 1),
```

The ternary is redundant.

Reply:
Gosh, my Bad, yeah!